src/channels.rs

```diff
@@ -25,7 +25,7 @@ impl MiEI {
         Ok(())
     }
 
-    pub fn get_role_id(&self, role_name: &str) -> Vec<RoleId> {
+    pub fn get_role_id(&self, role_name: &str) -> Vec<(String,RoleId)> {
```

Comment:
Try to return `Vec<(&str, RoleId)>` instead of cloning the strings. You might have to deal with lifetimes but it's faster and allocates less.